### DISCUSSION TITLE
Is it a Bug？ The containerID's PID is not in linux(centos)

### DISCUSSION THREAD

**Discussion by qmjun2023** [moby/moby#49088] [Q&A]
Hi，I am puzzled by a trouble;

My docker's version is 19.03.13, it runs in cenos 3.10.0-1160.el7.x86_64
I have a container, but it is dead(i don't know how and when it dead)
I use the  ‘docker ps' ,I can see the containerID ,the POD name infor; Then  I use the 'docker inspect containerid | grep Pid',  I also can get the Pid;

but I use the 'ps -ef | grep Pid' ,it is not exist; i check many times,it is true ,the Pid is not exist; 
there are also some infos in /var/log/messages, just like  that containerid can't be killed by signal 15

Is it a docker's bug ? and which version is is be fixed?

**Comment by thaJeztah** [comment]
This is quite likely after ((for some reason) the docker daemon missed an "exit" event from containerd. When dockerd starts a container through containerd, it monitors for events returned by containerd. There have been cases where no event was sent, for example,  when a container was killed by the kernel's OOM-killer (system out of memory), or if the container exited immediately after it was created. Most (if not all) of those should be fixed in current versions of docker and containerd.

here's one ticket related to this;

- https://github.com/moby/moby/issues/32827


:warning: I see you're running a _very_ old version of docker that reached End Of Life (EOL) over 5 years ago and is no longer maintained. It also looks like you're running on CentOS 7, which reached EOL as well (also no longer maintained), so if possible I'd highly recommend updating your system to  actively maintained versions of both, because there's various unpatched vulnerabilities in old versions of docker, containerd, and runc.

**Comment by qmjun2023** [comment]
thanks,i get it
